### PR TITLE
python3Packages.torchmetrics: 1.7.4 -> 1.8.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Diff: https://github.com/Lightning-AI/torchmetrics/compare/refs/tags/v1.7.4...refs/tags/v1.8.0
Changelog: https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.8.0

cc @SomeoneSerge

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 427836`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16479774745


---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 30 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 30 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427836`
Commit: `348368c5c3d330d165dda1321ff5d5721a918e84`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427836`
Commit: `348368c5c3d330d165dda1321ff5d5721a918e84`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 427836`
Commit: `348368c5c3d330d165dda1321ff5d5721a918e84`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>python312Packages.fastmri</li>
    <li>python312Packages.fastmri.dist</li>
    <li>python312Packages.finetuning-scheduler</li>
    <li>python312Packages.finetuning-scheduler.dist</li>
    <li>python312Packages.gluonts</li>
    <li>python312Packages.gluonts.dist</li>
    <li>python312Packages.lightning</li>
    <li>python312Packages.lightning.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.pytorch-lightning</li>
    <li>python312Packages.pytorch-lightning.dist</li>
    <li>python312Packages.torchmetrics</li>
    <li>python312Packages.torchmetrics.dist</li>
    <li>python312Packages.whisperx</li>
    <li>python312Packages.whisperx.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>whisperx.dist (python313Packages.whisperx.dist)</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Review by kirillrdy — APPROVED** [review]
